app/assets/javascripts/admin/modules/document-history.js

```diff
@@ -0,0 +1,45 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function DocumentHistory (module) {
+    this.module = module
+  }
+
+  DocumentHistory.prototype.init = function () {
+    this.setupEventListeners()
+  }
+
+  DocumentHistory.prototype.setupEventListeners = function () {
+    var documentHistoryTab = document.querySelector('#history_tab')
+
+    document.addEventListener('click', function (e) {
```

Comment:
This is dangerous as you're listening to click events on the whole document. I think what you need is click listeners on the individual links.

Reply:
The problem i found is if i scope it to the links is you end up in a bit of an infinite loop. The older and newer links get re-rendered when the component is passed back via html. You have to readd them, but you run into the same issue next time you click one of the links.

There's probs a simple way round this, but i'm not overly sure what the right thing to do is. 

Maybe just add it to the history tab? Or i guess just grab the html after this 

<img width="315" alt="image" src="https://user-images.githubusercontent.com/42515961/208706956-ee027fad-d6c5-4498-b017-a396f9b0d3df.png">

and render that?

Reply:
I think you should be fine to add the click handler to a parent DOM element, without needing to add it to the entire document.

For example, I'm assuming the element at `this.module` is a parent of `#history_tab`. So while we replace the contents of `#history_tab`, the parent `this.module` element persists so will keep its event listeners.

```suggestion
    this.module.addEventListener('click', function (e) {
```

Reply:
Thanks! Have sorted